packages/engine/src/vector-store/turbopuffer/filter.ts

```
@@ -0,0 +1,287 @@
+        // $exists: false -> use Eq with null (field doesn't exist if it is null)
+        return value ? [field, "NotEq", null] : [field, "Eq", null];
+
+      case "$all":
+        // $all is not directly supported, simulate with AND + IN conditions
+        if (!Array.isArray(value) || value.length === 0) {
+          throw new Error("$all operator requires a non-empty array");
+        }
+
+        const allConditions = value.map(
+          (item) => [field, "In", [this.normalizeValue(item)]] as Filter,
+        );
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)